### DISCUSSION TITLE
Forecast Information

### DISCUSSION THREAD

**Discussion by manapaap** [blaylockbk/Herbie#214] [Q&A]
Hi, I'm just getting started with using Herbie to download NWP data, so this might not be a well-formed question. 

Is there any means of querying the time step and spatial resolution of a model through Herbie? For instance, if I download GFS data, is there any means of knowing that it is on a 1 hour time step and that its spatial resolution drops from 28 km to 70 km after 7 days? Also, is there any way of querying fxx for < 1 hour for models such as HRRR which may have that data?

**Comment by blaylockbk** [comment]
Hi @manapaap, 

I hope you find Herbie helpful. Thanks for the questions.

> Is there any means of querying the time step and spatial resolution of a model through Herbie? For instance, if I download GFS data, is there any means of knowing that it is on a 1 hour time step and that its spatial resolution drops from 28 km to 70 km after 7 days?

Querying information about a model's resolution and timestep and forecast length are purposely not included in Herbie; maintaining a library of that info would be difficult to maintain as models change. However, I have included references for each of the models to help you discover those details for yourself. For example, [Herbie's GFS model template file](https://github.com/blaylockbk/Herbie/blob/main/herbie/models/gfs.py) has several links 

https://github.com/blaylockbk/Herbie/blob/f48ba8c1ec34829195b3673fed00e26f57dd5bbf/herbie/models/gfs.py#L8-L13

Each model's output is structured a little differently. In my experience working with new model data, it takes some time to figure out how the model output is organized; but keep at it and you will eventually understand the model output structure better.

<br><br>

> Also, is there any way of querying fxx for < 1 hour for models such as HRRR which may have that data?

Yes, refer to https://github.com/blaylockbk/Herbie/discussions/67. In short, the subhourly HRRR data is packed in hourly files. When you request forecast hour 1 `fxx=1`, you will find it has data at 15 min, 30 min, 45 min, and 60 min forecast. For example, in the index file https://noaa-hrrr-bdp-pds.s3.amazonaws.com/hrrr.20230718/conus/hrrr.t03z.wrfsubhf01.grib2.idx you will find it contains these GRIB messages for 2-m temperature

```
14:9469518:d=2023071803:TMP:2 m above ground:15 min fcst:
63:50631298:d=2023071803:TMP:2 m above ground:30 min fcst:
112:91096232:d=2023071803:TMP:2 m above ground:45 min fcst:
161:131312922:d=2023071803:TMP:2 m above ground:60 min fcst:
```
